src/components/pages/confirm/membership/index.tsx

```diff
@@ -185,28 +187,64 @@ export const ConfirmMembership: React.FC<ConfirmMembershipProps> = ({
         </>
       ) : (
         <>
+          {currentState.matches('pending') && <p>Setting up your account...</p>}
           <Header
             heading={<>Thank you so much for joining egghead! </>}
             primaryMessage={
               <>
-                <Callout>
-                  <IconMail className="p-3 rounded-full dark:bg-rose-500 dark:text-white bg-rose-100 text-rose-500" />
-                  <p className="text-lg">
-                    Please check your inbox to{' '}
-                    <strong>confirm your email address</strong>.
-                  </p>
-                </Callout>
-                <p className="text-lg">
-                  We've charged your credit card{' '}
-                  <strong className="dark:text-yellow-300 text-rose-500">
-                    ${session.amount} for an egghead membership
-                  </strong>{' '}
-                  and sent an email to{' '}
-                  <strong className="dark:text-yellow-300 text-rose-500">
-                    {session.email}
-                  </strong>{' '}
-                  so you can log in and access your membership.
-                </p>
+                {currentState.matches('pollingExpired') && (
+                  <>
+                    <Callout>
+                      <IconMail className="p-3 rounded-full dark:bg-rose-500 dark:text-white bg-rose-100 text-rose-500" />
+                      <p className="text-lg">
+                        Please check your inbox to{' '}
+                        <strong>confirm your email address</strong>.
+                      </p>
+                    </Callout>
+                    <p className="text-lg">
+                      We've charged your credit card{' '}
+                      <strong className="dark:text-yellow-300 text-rose-500">
+                        ${session.amount} for an egghead membership
+                      </strong>{' '}
+                      and sent an email to{' '}
+                      <strong className="dark:text-yellow-300 text-rose-500">
+                        {session.email}
+                      </strong>{' '}
+                      so you can log in and access your membership.
+                    </p>
+                  </>
+                )}
+                {currentState.matches('authTokenRetrieved') && (
```

Comment:
This is the happy path. We retrieved and stored their auth token, they are now ready to immediately view paid content 🎉